connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.ColumnIOFactory;
+import org.apache.parquet.io.MessageColumnIO;
+import org.apache.parquet.io.RecordReader;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+import org.teiid.file.VirtualFile;
+import org.teiid.file.VirtualFileConnection;
+import org.teiid.language.LanguageObject;
+import org.teiid.metadata.RuntimeMetadata;
+import org.teiid.translator.DataNotAvailableException;
+import org.teiid.translator.Execution;
+import org.teiid.translator.ExecutionContext;
+import org.teiid.translator.TranslatorException;
+
+
+public class BaseParquetExecution implements Execution {
+    @SuppressWarnings("unused")
+    protected ExecutionContext executionContext;
+    @SuppressWarnings("unused")
+    protected RuntimeMetadata metadata;
+    protected VirtualFileConnection connection;
+    protected boolean immutable;
+
+    // Execution state
+    protected ParquetQueryVisitor visitor = new ParquetQueryVisitor();
+    private VirtualFile[] parquetFiles;
+    private AtomicInteger fileCount = new AtomicInteger();
+    private ParquetFileReader reader;
+    private MessageType schema;
+    private RecordReader<Group> rowIterator;
+    private MessageColumnIO columnIO;
+    private long pageRowCount;
+
+    public BaseParquetExecution(ExecutionContext executionContext,
+                                RuntimeMetadata metadata, VirtualFileConnection connection, boolean immutable) {
+        this.executionContext = executionContext;
+        this.metadata = metadata;
+        this.connection = connection;
+        this.immutable = immutable;
+    }
+
+    public void visit(LanguageObject command) throws TranslatorException {
+        this.visitor.visitNode(command);
+
+        if (!visitor.exceptions.isEmpty()) {
+            throw visitor.exceptions.get(0);
+        }
+    }
+
+    @Override
+    public void execute() throws TranslatorException {
+        String path = this.visitor.getParquetPath();
+        if(this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONING_SCHEME) != null && this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONING_SCHEME).equals("directory")){
+            path += getDirectoryPath(this.visitor.getColumnValues(), this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS));
+        }
+        this.parquetFiles = VirtualFileConnection.Util.getFiles(path, this.connection, true);
+        VirtualFile nextParquetFile = getNextParquetFile();
+        if (nextParquetFile != null) {
+            this.rowIterator = readParquetFile(nextParquetFile);
+        }
+    }
+
+    private String getDirectoryPath(HashMap<String, String> columnValues, String partitionedColumns) {
+      String[] partitionedColumnsArray = getPartitionedColumns(partitionedColumns);
```

Comment:
Are you basing your example off of default spark partitioning?  We've also seen the convention where the directory structure contains the name and value:

dir/dim1=value1/dim2=value2/*

vs here we have

dir/value1/value2/*

 Do you seen any taxonomy or template for how the partitioning scheme is capture in the hive metastore?

Reply:
The spark used the "=" i.e of the form dir/dim1=value1/dim2=value2/*,  according to https://spark.apache.org/docs/2.4.0/sql-data-sources-parquet.html.

What I saw from a link, in hive partitioning looks like dir/value1/value2/*, see https://data-flair.training/blogs/apache-hive-partitions/ 

Then I checked a few pages today, one of then shows that it uses the same pattern dir/dim1=value1/dim2=value2/*, see https://www.guru99.com/hive-partitions-buckets-example.html

I think this dir/dim1=value1/dim2=value2/* should be considered, what do you say?

Reply:
> I think this dir/dim1=value1/dim2=value2/* should be considered, what do you say?

Yes it should be considered.  From what I can gather that is the dominant paradigm.